src/EnergyPlus/FaultsManager.cc

```diff
@@ -218,394 +277,722 @@ namespace FaultsManager {
+					ErrorsFound = true;
+				}
+			}
 
-					FaultsFouledAirFilters( jFaultyAirFilter ).FaultType = cFault1;
-					FaultsFouledAirFilters( jFaultyAirFilter ).FaultTypeEnum = iFaultTypeEnums( i );
-					FaultsFouledAirFilters( jFaultyAirFilter ).Name = cAlphaArgs( 1 );
+			// offset - degree of fault
+			FaultsChillerSWTSensor( jFault_ChillerSWT ).Offset = rNumericArgs( 1 );
+			
+			// Chiller type
+			FaultsChillerSWTSensor( jFault_ChillerSWT ).ChillerType = cAlphaArgs( 4 );
+			if ( lAlphaFieldBlanks( 4 ) ) {
+				ShowSevereError( cFaultCurrentObject + " = \"" + cAlphaArgs( 1 ) + "\" invalid " + cAlphaFieldNames( 4 ) + " = \"" + cAlphaArgs( 4 ) + "\" blank." );
+				ErrorsFound = true;
+			}
 
-					// Informatin of the fan associated with the fouling air filter
-					FaultsFouledAirFilters( jFaultyAirFilter ).FaultyAirFilterFanType = cAlphaArgs( 2 );
-					FaultsFouledAirFilters( jFaultyAirFilter ).FaultyAirFilterFanName = cAlphaArgs( 3 );
+			// Chiller name
+			FaultsChillerSWTSensor( jFault_ChillerSWT ).ChillerName = cAlphaArgs( 5 );
+			if ( lAlphaFieldBlanks( 5 ) ) {
+				ShowSevereError( cFaultCurrentObject + " = \"" + cAlphaArgs( 1 ) + "\" invalid " + cAlphaFieldNames( 5 ) + " = \"" + cAlphaArgs( 5 ) + "\" blank." );
+				ErrorsFound = true;
+			}
 
-					// Check whether the specified fan exsits in the fan list
-					if ( FindItemInList( cAlphaArgs( 3 ), Fans::Fan, &Fans::FanEquipConditions::FanName ) != 1 ) {
-						ShowSevereError( cFault1 + " = \"" + cAlphaArgs( 1 ) + "\" invalid " + cAlphaFieldNames( 3 ) + " = \"" + cAlphaArgs( 3 ) + "\" not found." );
-						ErrorsFound = true;
+			// Chiller check
+			{ auto const SELECT_CASE_VAR( FaultsChillerSWTSensor( jFault_ChillerSWT ).ChillerType );
+				
+				if( SameString( SELECT_CASE_VAR, "Chiller:Electric" ) ) {
+					// Read in chiller is not done yet
+					if ( PlantChillers::GetElectricInput ) {
+						PlantChillers::GetElectricChillerInput();
+						PlantChillers::GetElectricInput = false;
 					}
-
-					// Assign fault index to the fan object
-					for ( int FanNum = 1; FanNum <= Fans::NumFans; ++FanNum ) {
-						if ( SameString( Fans::Fan( FanNum ).FanName, cAlphaArgs( 3 ) ) ) {
-							Fans::Fan( FanNum ).FaultyFilterFlag = true;
-							Fans::Fan( FanNum ).FaultyFilterIndex = jFaultyAirFilter;
-							break;
+					// Check whether the chiller name and chiller type match each other
+					if ( FindItemInList( FaultsChillerSWTSensor( jFault_ChillerSWT ).ChillerName, PlantChillers::ElectricChiller.ma( &PlantChillers::ElectricChillerSpecs::Base ) ) <= 0 ) {
+						ShowSevereError( cFaultCurrentObject + " = \"" + cAlphaArgs( 1 ) + "\" invalid " + cAlphaFieldNames( 5 ) + " = \"" + cAlphaArgs( 5 ) + "\" not found." );
+						ErrorsFound = true;
+					} else {
+					// Link the chiller with the fault model
+						for ( int ChillerNum = 1; ChillerNum <= PlantChillers::NumElectricChillers; ++ChillerNum ) {
```

Comment:
@rongpengzhang Why search the chiller list again here?  FindItemInList above has already searched for the matching chiller, so you could save that result as ChillerNum then use it in the if and use it directly here.  Similar for the other chillers.